sarc/traces.py

```diff
@@ -0,0 +1,35 @@
+from contextlib import contextmanager
+
+from opentelemetry.trace import Status, StatusCode, get_tracer
+
+
+# context manager to manage traces & span,
+# and catch exceptions (in span) without breaking the whole program execution
+@contextmanager
+def using_trace(
+    tracer_name: str, span_name: str, exception_types=[Exception]
+):  # pylint: disable=dangerous-default-value
+    """
+    Context manager to manage traces & span,
+    and catch exceptions (in span) without breaking the whole program execution.
+        Parameters:
+                tracer_name (str): name of the tracer (can be shared between different spans)
+                span_name (str): name of the span
+                exception_types ([type,..]): types of exceptions to catch, other types will be raised
+                    by default, `Exception` is caught so that nothing would be raised
```

Comment:
```suggestion
                exception_types ([type,..]): types of exceptions to catch, other types will be raised.
                    By default, `Exception` is caught so that nothing would be raised
```

Reply:
docstring updated in https://github.com/mila-iqia/SARC/pull/109/commits/c4a3811d5f299b773f413106a9db663b001a4e68